backend/corpora/dutchnewspapers/dutchnewspapers_public.py

```diff
@@ -223,19 +213,11 @@ def fields(self):
             description='Whether the item is an article, advertisment, etc.',
             csv_core=True,
             es_mapping={'type': 'keyword'},
-            extractor=XML(tag='subject',
-                                  toplevel=True,
-                                  recursive=True,
-                                  multiple=False,
-                                  secondary_tag={
-                                      'tag': 'recordIdentifier',
-                                      'match': 'id'
-                                  },
-                                  external_file={
-                                      'xml_tag_toplevel': 'DIDL',
-                                      'xml_tag_entry': 'dcx'
-                                  }
-                                  ),
+            extractor=XML(
+                lambda metadata: Tag('recordIdentifier', string=metadata['id']),
+                SiblingTag('subject'),
+                external_file=True
```

Comment:
I mean, I agree that a complex yet untested corpus like this is broken by design, but this is already a huge PR so I would rather not stretch the scope any further.

Since an update to the corpus has been in progress for quite some time now, I had assumed that that process includes writing some unit tests _before_ starting a huge index job. I didn't want to spend too much time on this corpus as I assumed it would lead to double work.